app/views/govuk_publishing_components/components/_table.html.erb

```diff
@@ -2,46 +2,72 @@
   caption ||= nil
   head ||= []
   rows ||= []
-  first_cell_is_header ||=false
+  first_cell_is_header ||= false
   caption_classes ||= nil
-  sortable ||=false
+  sortable ||= false
+  filterable ||= false
+  label ||= t("components.table.filter_label")
 %>
 
-<%= GovukPublishingComponents::AppHelpers::TableHelper.helper(self, caption, {
-    sortable: sortable,
-    caption_classes: caption_classes
-  }) do |t| %>
+<% @table = capture do %>
+  <%= GovukPublishingComponents::AppHelpers::TableHelper.helper(self, caption, {
+      sortable: sortable,
+      filterable: filterable,
+      caption_classes: caption_classes
+    }) do |t| %>
 
-  <% if head.any? %>
-    <%= t.head do %>
-      <% head.each_with_index do |item, cellindex| %>
-        <%= t.header item[:text], {
-          format: item[:format],
-          href: item[:href],
-          data_attributes: item[:data_attributes],
-          sort_direction: item[:sort_direction]
-        } %>
+    <% if head.any? %>
+      <%= t.head do %>
+        <% head.each_with_index do |item, cellindex| %>
+          <%= t.header item[:text], {
+            format: item[:format],
+            href: item[:href],
+            data_attributes: item[:data_attributes],
+            sort_direction: item[:sort_direction]
+          } %>
+        <% end %>
       <% end %>
     <% end %>
-  <% end %>
 
-  <%= t.body do %>
-    <% rows.each_with_index do |row, rowindex| %>
-      <%= t.row do %>
-        <% row.each_with_index do |cell, cellindex| %>
-          <% if cellindex == 0 && first_cell_is_header %>
-            <%= t.header cell[:text], {
-              scope: "row",
-              format: cell[:format]
-            } %>
-          <% else %>
-            <%= t.cell cell[:text], {
-              format: cell[:format]
-            } %>
+    <%= t.body do %>
+      <% rows.each_with_index do |row, rowindex| %>
+        <%= t.row do %>
+          <% row.each_with_index do |cell, cellindex| %>
+            <% if cellindex == 0 && first_cell_is_header %>
+              <%= t.header cell[:text], {
+                scope: "row",
+                format: cell[:format]
+              } %>
+            <% else %>
+              <%= t.cell cell[:text], {
+                format: cell[:format]
+              } %>
+            <% end %>
           <% end %>
         <% end %>
       <% end %>
     <% end %>
   <% end %>
+<% end %>
+
+<% if filterable %>
+  <div data-module="table-filter">
+    <div class="js-gem-c-table__filter govuk-!-display-none" aria-live="assertive">
```

Comment:
I think I'm getting closer to the problem with the screenreader announcement - this `aria-live` attribute needs to be on the`<p id="filterCount">`.

Reply:
Doing so seems to improve screen reader announcement, but not to 100%. I think part of the problem is that most screenreaders will only announce the hidden text if it changes, which means you could continue to type something that matches nothing and only receive one announcement. But that might be okay.